src/ui/components/delegation/DelegateCard.tsx

```diff
@@ -72,10 +81,14 @@ export function DelegateCard({
 }) {
   const { name, address, manifesto } = delegate;
   const roles = extractRole(address, state);
+  const { preview, truncated } = useMemo(
```

Comment:
Would it make sense to use https://react.dev/reference/react/memo on the whole component?

Reply:
Maybe after we memoize the props as well?! currently, I believ memo won't help much, since the state can change often?!